lib/msf/core/payload_generator.rb

```diff
@@ -171,12 +171,12 @@ def add_shellcode(shellcode)
     # This method takes a payload module and tries to reconcile a chosen
     # arch with the arches supported by the module.
     # @param mod [Msf::Payload] The module class to choose an arch for
-    # @return [String] String form of the Arch if a valid arch found
+    # @return [String] String form of the arch if a valid arch found
     # @return [Nil] if no valid arch found
     def choose_arch(mod)
       if arch.blank?
         @arch = mod.arch.first
-        cli_print "No Arch selected, selecting Arch: #{arch} from the payload"
+        cli_print "[-] No --arch selected, selecting arch: #{arch} from the payload"
```

Comment:
So far, I think `msfvenom` is the only user of `PayloadGenerator`, but any other users probably won't have `--arch` or `msfvenom` options.

Reply:
How likely is that to change?
What to me put it back to how it was?

Reply:
My brain sorta ignores the `--` if it isn't relevant, so I don't have a preference.

Reply:
I have a preference: the payload generator library should not reference command-line arguments from `msfvenom`. It's counter to sustainable design.